src/web/app/components/question-submission-form/question-submission-form.component.ts

```diff
@@ -197,4 +196,13 @@ export class QuestionSubmissionFormComponent implements OnInit {
     return this.feedbackResponseService.isFeedbackResponseDetailsEmpty(
         this.model.questionType, responseDetails);
   }
+
+  /**
+   * Updates validity of all responses in a question.
+   */
+  updateValidity(isValid: boolean): void {
+    this.model.recipientSubmissionForms.forEach((model: FeedbackResponseRecipientSubmissionFormModel) => {
+      model.isValid = isValid;
```

Comment:
I think I need to emphasize this again :(

This is not the correct way to change the value **passed** by the **parent** component. In other word, you do **NOT** own the model so you should not change it directly. Instead, please use event to emit the changes.

I think we need pass this knowledge down:
- All these efforts are to make the parent component who **own** the model to become a single point of truth
   - The parent component is free to make any change. It can also **ignore** the event omit by the child component.
   - Image that if child components make the change directly (like in this case and it works), how do you trace the control flow? In other word, you are breaking the contract that the `@Input()` annotation tells. It should be read-only and input property only.
   - (NOT relevant to this but useful to know)If you want to make the change to `@Input()` property, create a new variable and sync it between the input property when its value changes (see `StudentList` for example).

Reply:
In case you are interested, using events might cause very deep passing of event handler from child all the way to root. Therefore, we have a more proper solution called [Redux](https://redux.js.org/ mostly used for React app) (or [NGRX](https://ngrx.io/) for Angular). We will see if we can leverage it in due course (we might not in that state yet, see below). 

> In general, use Redux when you have reasonable amounts of data changing over time, you need a single source of truth, and you find that approaches like keeping everything in a top-level React component's state are no longer sufficient.

> It's intended to help answer the question "When did a certain slice of state change, and where did the data come from?", with predictable behavior.

Reply:
Could I use the `formModelChange` emitter to fire the event or should I create a new `@Output`?